docs/design/agent-and-add-in-overview.md

```diff
@@ -0,0 +1,243 @@
+                {
+                    "id": "fillcolor",
+                    "type": "executeDataFunction",
+                }
+            ]
+        }
+    ]
+]
+```
+
+Second, create a declarative agent object that identifies the file containing the detailed configuration of the agent. The following is an example.
+
+```json
+"copilotAgents": {
+    "declarativeAgents": [
+        {
+        "id": "ContosoCopilotAgent",
+        "file": "declarativeAgent.json"
+        }
+    ]
+}
+```
+
+The reference documentation for the manifest JSON is at [Microsoft 365 app manifest schema reference](/microsoft-365/extensibility/schema/).
+
+### Declarative agent configuration
+
+The agent configuration file includes instructions for the agent and specifies one or more API plugin configuration files that will contain the detailed configuration of the agent's actions. The following is an example. Note the following about this JSON.
+
+- The conversation starter appears in the chat canvas of Copilot.
+- The `"actions.id"` property in this file is the collective ID of all the functions in the file specified in `"actions.file"`. It doesn't have to match the `"actions.id"` in the manifest.
+
+```json
+{
+    "name": "Excel Add-in + Agent",
+    "description": "Agent for working with Excel cells.",
+    "instructions": "You are an agent for working with an add-in. You can work with any cells, not just a well-formatted table.",
+    "conversation_starters": [
+        {
+            "title": "Change cell color",
+            "text": "I want to change the color of cell B2 to orange"
+        }
+    ],
+    "actions": [
+        {
+            "id": "localExcelPlugin",
+            "file": "Excel-API-local-plugin.json"
+        }
+    ]
+}
+```
+
+The reference documentation for declarative agents is at [Declarative agent schema 1.3 for Microsoft 365 Copilot](/microsoft-365-copilot/extensibility/declarative-agent-manifest-1.3).
+
+### Copilot API plugin configuration
+
+The API plugin configuration file specifies the "functions" of the plugin in the sense of agent actions, not JavaScript functions, including the instructions for the action. It also configures the JavaScript runtime for Copilot. The following is an example. About this JSON, note the following:
+
+- The `"functions.name"` must match the `"extensions.runtimes.actions.id"` property in the add-in manifest.
+- The `"reasoning.description"` and `"reasoning.instructions"` refer to a JavaScript function, not a REST API.
+- The `"runtimes.run_for_functions"` array must include either the same string as `"functions.name"` or a wildcard string that matches it.
+- The `"runtimes.spec.local_endpoint"` property is new and isn't yet in the main reference documentation for the API plugins schema. See below for more about it. In this case, it specifies that the JavaScript function that is associated with the "fillcolor" string is available in an Office add-in, rather than in some REST endpoint.
```

Comment:
```suggestion
- The `"runtimes.spec.local_endpoint"` property is new and isn't yet in the main reference documentation for the API plugins schema. See below for more about it. In this case, it specifies that the JavaScript function that is associated with the "fillcolor" string is available in an Office Add-in, rather than in some REST endpoint.
```